### PR TITLE
MySQL / Postgres support with dynamic passwords for docker-compose

### PR DESCRIPTION
**Pull request by chriske**
#724 

This Pull request implements the enhancement what we talked about in that issue above.

I splitted up the compose file's database part into two separate files. In this way we're able to support both databases out of the box. I adjusted the setup script, now it needs to know the selected database runtime to be able to generate the connection URL properly. I also added dynamic password generation for every password in the environment file, and the microcks-service-account uuid as well.

I had some difficulities with the PowerShell script, so I completely removed it. I made a wrapper container what can run the setup script inside of it, so the setup script is now platform independent.

I made the changes to the readme also.

Please check it, and we can discuss the changes!

Thanks!

### CODE REVIEW

File: distro/docker-compose/Readme.md [hunk 1]
```diff
@@ -13,50 +13,95 @@ Here is the port mapping:
...
+If you generated a config for MySQL:
+docker-compose -f docker-compose.keycloak.yml -f docker-compose.microcks.yml -f docker-compose.apicurio.yml -f docker-compose-as-mysql.yml up
+
+If you generated a config for PostgreSQL:
+docker-compose -f docker-compose.keycloak.yml -f docker-compose.microcks.yml -f docker-compose.apicurio.yml -f docker-compose-as-postgre.yml up
+
 ```
 
+Please do not mix up those commands! If you want to switch between databases, you have to clear the already existing volumes and configs.
+
```
Comment (EricWittmann) [inline distro/docker-compose/Readme.md]: Can you provide the commands to do that?  Or perhaps include that in the reset script?

**Comment by EricWittmann** [comment]
This is great!  Nice job.  The only suggestion I have is to perhaps generate a **run** script of some sort (e.g. a simple bash script) with the correct `docker-compose` command in it.  That way it will be harder for the user to run the wrong command (you have a warning in the docs about it, which is good).

I'm happy to merge this PR without that addition - it's just a suggestion. :)

File: distro/docker-compose/Readme.md (hunk 1 above)
Comment (chriske) [reply distro/docker-compose/Readme.md]: done :)

**Comment by chriske** [comment]
Thanks! I made the suggested changes in the next commit. The start script was a good idea, thx :)

I didn't included the docker volume deletion command in the reset-env script, because I didn't want to people accidentally delete their volumes :) So I just extended that part of the Readme. 

Let me know if this is what you suggested :)

**Comment by EricWittmann** [comment]
Looks great - nice work.

**Comment by chriske** [comment]
@EricWittmann I have also a question. Last time when I created the compose files, there was an issue with Microcks integration. I wasn't able to mock anything, because the examples didn't appeared in Microcks just the endpoints. Now I tested it again with the PetStore example. Now I was able to mock the endpoint where I can get a list of Pets, and that worked fine. But the other endpoint with the petID doesn't. I tried with a lot of examples, checked your video, and Microcks showed me time-to-time that there are no examples at all. Maybe I made something wrong, or there is an issue with the integration right now?  I'm hoping that not the docker config is the reason. I also checked the logs, but I didn't noticed any errors. Thanks!

**Comment by EricWittmann** [comment]
Which specific endpoints work vs. don't work in the Pet Store 3.0 sample?

**Comment by chriske** [comment]
/pets is working, and /pets/{id} doesn't.

image

**Comment by chriske** [comment]
Also here is the relevant part from the logs. As you can see, it can't find the responses:

`apicurio-studio-api_1   | 2019-07-30 14:21:46,154 DEBUG [io.apicurio.studio.shared.config.Configuration] (default task-11) Config Property: APICURIO_MICROCKS_CLIENT_ID/apicurio.hub.microcks.clientId = microcks-serviceaccount
apicurio-studio-api_1   | 2019-07-30 14:21:46,155 DEBUG [io.apicurio.studio.shared.config.Configuration] (default task-11) Config Property: APICURIO_MICROCKS_CLIENT_SECRET/apicurio.hub.microcks.clientSecret = 723d4d6a-c222-4e0d-bb82-03ac0b401695
microcks                | 14:21:46.168 DEBUG 1 --- [80-exec-10] i.g.m.web.UploadArtifactController       : Content type of open-api-contract.yml is application/octet-stream
microcks                | 14:21:46.174 DEBUG 1 --- [80-exec-10] i.g.microcks.service.ServiceService      : Service [storeExample, 1.0.0] exists ? true
microcks                | 14:21:46.189  INFO 1 --- [80-exec-10] i.g.microcks.service.ServiceService      : Having imported 1 services definitions into repository
apicurio-studio-api_1   | 2019-07-30 14:21:46,190 INFO  [io.apicurio.hub.api.microcks.MicrocksConnector] (default task-11) Microcks mocks have been created/updated for storeExample:1.0.0
apicurio-studio-api_1   | 2019-07-30 14:21:46,190 DEBUG [io.apicurio.studio.shared.config.Configuration] (default task-11) Config Property: APICURIO_MICROCKS_API_URL/apicurio.hub.microcks.api = http://192.168.1.231:8900/api
apicurio-studio-api_1   | 2019-07-30 14:21:46,190 DEBUG [io.apicurio.hub.core.storage.jdbc.JdbcStorage] (default task-11) Inserting a 'command' content row for: 1
microcks                | 14:21:49.002 DEBUG 1 --- [080-exec-1] i.g.m.web.KeycloakConfigController       : Returning 'microcks' realm config, for http://192.168.1.231:8090/auth
microcks                | 14:21:49.072 DEBUG 1 --- [080-exec-3] i.g.m.web.KeycloakConfigController       : Returning 'microcks' realm config, for http://192.168.1.231:8090/auth
microcks                | 14:21:49.113 DEBUG 1 --- [080-exec-4] i.g.m.web.VersionInfoController          : Returning '0.7.2-SNAPSHOT' version information
microcks                | 14:21:49.138 DEBUG 1 --- [080-exec-5] i.g.microcks.web.ServiceController       : Retrieving service with id storeExample:1.0.0
microcks                | 14:21:49.142 DEBUG 1 --- [080-exec-5] i.g.microcks.service.MessageService      : Found 1 request(s) for operation 5d4050be36d1a000010e5404-GET /pets
microcks                | 14:21:49.142 DEBUG 1 --- [080-exec-5] i.g.microcks.service.MessageService      : Found 1 response(s) for operation 5d4050be36d1a000010e5404-GET /pets
microcks                | 14:21:49.142 DEBUG 1 --- [080-exec-5] i.g.microcks.service.MessageService      : Emitting 1 request/response pair(s) as result
microcks                | 14:21:49.144 DEBUG 1 --- [080-exec-5] i.g.microcks.service.MessageService      : Found 1 request(s) for operation 5d4050be36d1a000010e5404-POST /pets
microcks                | 14:21:49.144 DEBUG 1 --- [080-exec-5] i.g.microcks.service.MessageService      : Found 1 response(s) for operation 5d4050be36d1a000010e5404-POST /pets
microcks                | 14:21:49.144 DEBUG 1 --- [080-exec-5] i.g.microcks.service.MessageService      : Emitting 1 request/response pair(s) as result
microcks                | 14:21:49.146 DEBUG 1 --- [080-exec-5] i.g.microcks.service.MessageService      : Found 0 request(s) for operation 5d4050be36d1a000010e5404-GET /pets/{id}
microcks                | 14:21:49.146 DEBUG 1 --- [080-exec-5] i.g.microcks.service.MessageService      : Found 0 response(s) for operation 5d4050be36d1a000010e5404-GET /pets/{id}
microcks                | 14:21:49.146 DEBUG 1 --- [080-exec-5] i.g.microcks.service.MessageService      : Emitting 0 request/response pair(s) as result
microcks                | 14:21:49.148 DEBUG 1 --- [080-exec-5] i.g.microcks.service.MessageService      : Found 0 request(s) for operation 5d4050be36d1a000010e5404-DELETE /pets/{id}
microcks                | 14:21:49.148 DEBUG 1 --- [080-exec-5] i.g.microcks.service.MessageService      : Found 0 response(s) for operation 5d4050be36d1a000010e5404-DELETE /pets/{id}
microcks                | 14:21:49.148 DEBUG 1 --- [080-exec-5] i.g.microcks.service.MessageService      : Emitting 0 request/response pair(s) as result
microcks                | 14:21:49.151 DEBUG 1 --- [080-exec-6] i.g.microcks.web.ServiceController       : Retrieving service with id storeExample:1.0.0
microcks                | 14:21:49.154 DEBUG 1 --- [080-exec-6] i.g.microcks.service.MessageService      : Found 1 request(s) for operation 5d4050be36d1a000010e5404-GET /pets
microcks                | 14:21:49.154 DEBUG 1 --- [080-exec-6] i.g.microcks.service.MessageService      : Found 1 response(s) for operation 5d4050be36d1a000010e5404-GET /pets
microcks                | 14:21:49.154 DEBUG 1 --- [080-exec-6] i.g.microcks.service.MessageService      : Emitting 1 request/response pair(s) as result
microcks                | 14:21:49.155 DEBUG 1 --- [080-exec-6] i.g.microcks.service.MessageService      : Found 1 request(s) for operation 5d4050be36d1a000010e5404-POST /pets
microcks                | 14:21:49.155 DEBUG 1 --- [080-exec-6] i.g.microcks.service.MessageService      : Found 1 response(s) for operation 5d4050be36d1a000010e5404-POST /pets
microcks                | 14:21:49.155 DEBUG 1 --- [080-exec-6] i.g.microcks.service.MessageService      : Emitting 1 request/response pair(s) as result
microcks                | 14:21:49.156 DEBUG 1 --- [080-exec-6] i.g.microcks.service.MessageService      : Found 0 request(s) for operation 5d4050be36d1a000010e5404-GET /pets/{id}
microcks                | 14:21:49.157 DEBUG 1 --- [080-exec-6] i.g.microcks.service.MessageService      : Found 0 response(s) for operation 5d4050be36d1a000010e5404-GET /pets/{id}
microcks                | 14:21:49.157 DEBUG 1 --- [080-exec-6] i.g.microcks.service.MessageService      : Emitting 0 request/response pair(s) as result
microcks                | 14:21:49.157 DEBUG 1 --- [080-exec-6] i.g.microcks.service.MessageService      : Found 0 request(s) for operation 5d4050be36d1a000010e5404-DELETE /pets/{id}
microcks                | 14:21:49.157 DEBUG 1 --- [080-exec-6] i.g.microcks.service.MessageService      : Found 0 response(s) for operation 5d4050be36d1a000010e5404-DELETE /pets/{id}
microcks                | 14:21:49.157 DEBUG 1 --- [080-exec-6] i.g.microcks.service.MessageService      : Emitting 0 request/response pair(s) as result
microcks                | 14:21:49.176 DEBUG 1 --- [080-exec-7] i.g.microcks.web.ResourceController      : Request resources for service storeExample:1.0
microcks                | 14:21:49.179 DEBUG 1 --- [080-exec-9] i.github.microcks.web.TestController     : Getting tests list for service storeExample:1.0, page 0 and size 20
`

**Comment by EricWittmann** [comment]
Before I dig into this, I wonder if @lbroudoux has any quick insight into things.

**Comment by lbroudoux** [comment]
Hi there !
Have you tried importing directly your specification into Microcks to see if missing endpoints are independant of integration of Apicurio ? Could you share your contract or give a way to reproduce it ? You mention a video and a PetStore example ...
I will be able to look at this on tomorrow

**Comment by EricWittmann** [comment]
@lbroudoux I believe he is using the Pet Store sample that is included in Apicurio Studio.  When you create a new API you have the option of starting with a few samples - one of them is Pet Store.  I can attach it here if that's more convenient!

**Comment by chriske** [comment]
@EricWittman yes, exactly! Thanks!

**Comment by lbroudoux** [comment]
Hi,
Just restart from the Pet Store sample on studio.apicur.io online instance and got some explanations.

First - I don't know if it's something that has been added in more recent version - I had to add an example for the `GET /pets` operation; just creating an array of pets as response body.

Then, I have to define example for the `GET /pets/{id}` operation:
* defining a pet as response example,
* creating the `id` path parameter,
* defining an example for `id` - this example should have the same name as response example.

However, this later point - example for `id` path parameter - cannot be done yet into Apicurio GUI and I have to add it using the `source` view. 

Here's the OpenAPI contract containing my additions : https://gist.github.com/lbroudoux/e56296de0617f29e7fd2c455e523b177 

Using this contract, I get the following results into Microcks with working mocks:

Capture d’écran 2019-07-31 à 13 04 25

Hope it will help!

**Comment by EricWittmann** [comment]
Thanks @lbroudoux  - I have pushed an update to Apicurio Studio with [your changes to the pet store sample](https://github.com/Apicurio/apicurio-studio/commit/02d9f0998fa63ba2828a800a170b42bef65d9e3c).  And we REALLY need to get parameter example editing working implemented. :(

**Comment by lbroudoux** [comment]
Oh yes ! Definitely. I think I may have some time this week. I'll try that !!

**Comment by EricWittmann** [comment]
I'd of course be happy for the contribution, but please don't think I was volunteering you for that work!!

**Comment by lbroudoux** [comment]
;-) I’ll be also happy to tell people I met they can do everything in
Apicurio and don’t need to switch to source anymore.

Le mer. 31 juil. 2019 à 16:27, Eric Wittmann <notifications@github.com> a
écrit :

**Comment by EricWittmann** [comment]
Is that (parameter examples) the only microcks-related reason to use the source tab now?

**Comment by lbroudoux** [comment]
Yes I think so. There's no other reason coming to my mind.

Le mer. 31 juil. 2019 à 19:18, Eric Wittmann <notifications@github.com> a
écrit :

**Pull request merged**